morpheus/_lib/tests/stages/test_triton_inference_stage.cpp

```diff
@@ -0,0 +1,337 @@
+        }
+
+        callback(new FakeInferResult({{"seq_ids", std::vector<int32_t>({0, 1, 2, 3, 4, 5, 6, 7, 8, 9})}}));
+
+        return triton::client::Error::Success;
+    }
+};
+
+class TestTritonInferenceStage : public morpheus::test::TestWithPythonInterpreter
+{
+  protected:
+    void SetUp() override
+    {
+        morpheus::test::TestWithPythonInterpreter::SetUp();
+        {
+            pybind11::gil_scoped_acquire gil;
+
+            // Initially I ran into an issue bootstrapping cudf, I was able to work-around the issue, details in:
+            // https://github.com/rapidsai/cudf/issues/12862
+            morpheus::CudfHelper::load();
+        }
+    }
+};
+
+cudf::io::table_with_metadata create_test_table_with_metadata(uint32_t rows)
+{
+    cudf::data_type cudf_data_type{cudf::type_to_id<int>()};
+
+    auto column = cudf::make_fixed_width_column(cudf_data_type, rows);
+
+    std::vector<int> data(rows);
+    std::iota(data.begin(), data.end(), 0);
+
+    cudaMemcpy(column->mutable_view().data<int>(),
+               data.data(),
+               data.size() * sizeof(int),
+               cudaMemcpyKind::cudaMemcpyHostToDevice);
+
+    std::vector<std::unique_ptr<cudf::column>> columns;
+
+    columns.emplace_back(std::move(column));
+
+    auto table = std::make_unique<cudf::table>(std::move(columns));
+
+    auto index_info   = cudf::io::column_name_info{""};
+    auto column_names = std::vector<cudf::io::column_name_info>({{index_info}});
+    auto metadata     = cudf::io::table_metadata{std::move(column_names), {}, {}};
+
+    return cudf::io::table_with_metadata{std::move(table), metadata};
+}
+
+TEST_F(TestTritonInferenceStage, SingleRow)
+{
+    cudf::data_type cudf_data_type{cudf::type_to_id<int>()};
+
+    const std::size_t count = 10;
+    const auto dtype        = morpheus::DType::create<int>();
+    auto buffer = std::make_shared<rmm::device_buffer>(count * dtype.item_size(), rmm::cuda_stream_per_thread);
+    std::vector<int> seq_ids({0, 1, 2, 3, 4, 5, 6, 7, 8, 9});
+    cudaMemcpy(buffer->data(), seq_ids.data(), count * sizeof(int), cudaMemcpyKind::cudaMemcpyHostToDevice);
+    auto tensors = morpheus::TensorMap();
+    tensors["seq_ids"].swap(morpheus::Tensor::create(buffer, dtype, {count, 1}, {}));
+    auto memory  = std::make_shared<morpheus::TensorMemory>(count, std::move(tensors));
+    auto table   = create_test_table_with_metadata(count);
+    auto meta    = morpheus::MessageMeta::create_from_cpp(std::move(table), 1);
+    auto message = std::make_shared<morpheus::MultiInferenceMessage>(meta, 0, count, memory);
+
+    auto triton_client           = std::make_unique<FakeTritonClient>();
+    auto triton_inference_client = std::make_unique<morpheus::TritonInferenceClient>(std::move(triton_client), "");
+    auto stage = morpheus::InferenceClientStage(std::move(triton_inference_client), "", false, {}, {});
+
+    auto on           = std::make_shared<mrc::coroutines::TestScheduler>();
```

Comment:
This is one giant block of code with no comments. Break it up into logical sections and add comments.